media/editor/dataDisplay.tsx

```diff
@@ -387,10 +389,14 @@ const DataPage: React.FC<IDataPageProps> = props => (
 	</div>
 );
 
-const generateRows = (props: IDataPageProps, fn: (offset: number) => React.ReactChild) => {
+const generateRows = (
+	props: IDataPageProps,
+	fn: (offset: number, isRowWithInsertDataCell: boolean) => React.ReactChild,
+) => {
 	const rows: React.ReactNode[] = [];
 	let row = (props.rowsStart - props.pageStart) / props.columnWidth;
-	for (let i = props.rowsStart; i < props.rowsEnd && i < props.fileSize; i += props.columnWidth) {
+	const lastRowIndex = props.fileSize - (props.fileSize % props.columnWidth);
```

Comment:
nit: modulo is relative slow, I usually prefer to floor and multiply :)

```suggestion
	const lastRowIndex = props.columnWidth * Math.floor(props.fileSize / props.columnWidth);
```